pyquil/quilatom.py

```diff
@@ -537,6 +540,37 @@ def __neg__(self) -> "Mul":
     def _substitute(self, d: Any) -> ExpressionDesignator:
         return self
 
+    def _simplify(self) -> quil_rs_expr.Expression:
+        rs_expression = _convert_to_rs_expression(self)
+        rs_expression.simplify()
+        return rs_expression
```

Comment:
Not changes requested, but for your awareness in case it comes up in the near future - expression simplification can be expensive.